piptools/scripts/compile.py

```diff
@@ -291,6 +305,27 @@ def _determine_linesep(
     help="Specify a package to consider unsafe; may be used more than once. "
     f"Replaces default unsafe packages: {', '.join(sorted(UNSAFE_PACKAGES))}",
 )
+@click.option(
+    "--build-distribution",
+    "build_distributions",
+    multiple=True,
+    type=click.Choice(("editable", "sdist", "wheel")),
+    help="Name of a distribution to install build dependencies for; may be used more than once. "
+    "Static dependencies declared in pyproject.toml will be included as well.",
+)
+@click.option(
+    "--all-build-distributions",
+    is_flag=True,
+    default=False,
+    help="Pin build dependencies requested by any of the build backend hooks. "
+    "Static dependencies declared in pyproject.toml will be included as well.",
+)
+@click.option(
+    "--only-build-distributions",
```

Comment:
Same here.

Reply:
Marking this as resolved since it is now hard to track in the code and I have made an attempt at addressing this.